### PR TITLE
LPD-34381 - Phone api (part2)

### PR DESCRIPTION
**Pull request by brbalazs**
https://liferay.atlassian.net/browse/LPD-34381

### CODE REVIEW

**Comment by brbalazs** [comment]
ci:test:sf

File: modules/apps/headless/headless-admin-user/headless-admin-user-impl/rest-openapi.yaml [hunk 1]
```diff
@@ -4845,7 +4849,89 @@ paths:
                                     $ref: "#/components/schemas/Organization"
                                 type: array
             tags: ["Organization"]
+    "/phones/by-external-reference-code/{externalReferenceCode}":
+        delete:
+            description:
+                Deletes the phone number using external reference code.
```
Comment (smottal) [inline modules/apps/headless/headless-admin-user/headless-admin-user-impl/rest-openapi.yaml:4855]: can you check the other description we used for website or emailaddress?

File: modules/apps/headless/headless-admin-user/headless-admin-user-impl/rest-openapi.yaml [hunk 2]
```diff
@@ -4845,7 +4849,89 @@ paths:
...
+                      type: string
+            responses:
+                204:
+                    content:
+                        application/json: {}
+                        application/xml: {}
+            tags: ["Phone"]
+        get:
+            description:
+                Retrieves the phone number using external reference code.
```
Comment (smottal) [inline modules/apps/headless/headless-admin-user/headless-admin-user-impl/rest-openapi.yaml:4871]: can you check the other description we used for website or emailaddress?

File: modules/apps/headless/headless-admin-user/headless-admin-user-impl/rest-openapi.yaml [hunk 3]
```diff
@@ -4845,7 +4849,89 @@ paths:
...
+                        application/json:
+                            schema:
+                                $ref: "#/components/schemas/Phone"
+                        application/xml:
+                            schema:
+                                $ref: "#/components/schemas/Phone"
+            tags: ["Phone"]
+        patch:
+            description:
+                Updates the phone number using external reference code.
```
Comment (smottal) [inline modules/apps/headless/headless-admin-user/headless-admin-user-impl/rest-openapi.yaml:4891]: can you check the other description we used for website or emailaddress?

File: modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/PhoneResourceImpl.java [hunk 4]
```diff
@@ -35,6 +40,28 @@
...
+
+		com.liferay.portal.kernel.model.Phone phone =
+			_phoneService.fetchPhoneByExternalReferenceCode(
+				externalReferenceCode, contextCompany.getCompanyId());
+
+		if (phone == null) {
+			throw new NoSuchPhoneException(
+				"Unable to find phone with external reference code " +
+					externalReferenceCode);
+		}
```
Comment (smottal) [inline modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/PhoneResourceImpl.java:60]: now you can use the get

File: modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/PhoneResourceImpl.java [hunk 5]
```diff
@@ -89,6 +116,23 @@ public Phone getPhone(Long phoneId) throws Exception {
...
```
Comment (smottal) [inline modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/PhoneResourceImpl.java:131]: now you can use the get

File: modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/PhoneResourceImpl.java [hunk 6]
```diff
@@ -112,12 +156,69 @@ public Page<Phone> getUserAccountPhonesPage(Long userAccountId)
 				PhoneUtil::toPhone));
 	}
 
+	@Override
+	public Phone patchPhone(Long phoneId, Phone phone) throws Exception {
+		com.liferay.portal.kernel.model.Phone servicePhone =
```
Comment (smottal) [inline modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/PhoneResourceImpl.java:161]: serviceBuilderPhone (please check in the other classes)

File: modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/PhoneResourceImpl.java [hunk 7]
```diff
@@ -112,12 +156,69 @@ public Page<Phone> getUserAccountPhonesPage(Long userAccountId)
...
+			_phoneService.fetchPhoneByExternalReferenceCode(
+				externalReferenceCode, contextCompany.getCompanyId());
+
+		if (servicePhone == null) {
+			throw new NoSuchPhoneException(
+				"Unable to find phone with external reference code " +
+					externalReferenceCode);
+		}
+
```
Comment (smottal) [inline modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/PhoneResourceImpl.java:197]: you can use the get

**Review by smottal — CHANGES_REQUESTED** [review]

**Review by smottal — APPROVED** [review]

**Comment by brbalazs** [comment]
ci:test:sf

**Comment by brbalazs** [comment]
ci:test:relevant

**Comment by brbalazs** [comment]
ci:test:relevant

**Comment by smottal** [comment]
ci:test:relevant

**Comment by smottal** [comment]
ci:test:relevant

**Comment by smottal** [comment]
ci:forward

**Pull request closed without merging**